tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/image/ops/TransformToGrayscaleOp.java

```diff
@@ -0,0 +1,141 @@
+import android.graphics.PointF;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.tensorflow.lite.support.image.ImageOperator;
+import org.tensorflow.lite.support.image.TensorImage;
+import org.tensorflow.lite.support.tensorbuffer.TensorBuffer;
+import org.tensorflow.lite.DataType;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import static org.tensorflow.lite.support.image.ColorSpaceType.GRAYSCALE;
+
+/**
+ * As a computation unit for processing images, it could transform a colored image to grayscale.
+ *
+ * The conversion is based on OpenCV RGB to GRAY conversion
+ * https://docs.opencv.org/master/de/d25/imgproc_color_conversions.html#color_convert_rgb_gray
+ *
+ * The 3 channels of the grayscale image are identical and one of them is used to create the TensorBuffer
+ *
+ */
+public class TransformToGrayscaleOp implements ImageOperator {
+
+    /**
+     * Creates a TransformToGrayscaleOp
+     *
+     * No parameters are used
+     */
+    public TransformToGrayscaleOp() {
+    }
+
+    /**
+     * Applies the defined transformation to grayscale and returns a TensorImage
+     *
+     * Because the 3 channels of the image are identical only one is used to create the TensorBuffer
+     *
+     * @param image input image.
+     * @return output image.
+     */
+    @Override
+    @NonNull
+    public TensorImage apply(@NonNull TensorImage image) {
+
+        int height = image.getBitmap().getHeight();
+        int width = image.getBitmap().getWidth();
+        Bitmap bmpGrayscale = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888);
+        Canvas canvas = new Canvas(bmpGrayscale);
+        Paint paint = new Paint();
+        // A matrix is created that will be applied to canvas to generate grayscale image
+        // The luminance of each pixel is calculated as the weighted sum of the 3 RGB values
+        // Y = 0.299R + 0.587G + 0.114B
+        float[] matrix = new float[]{0.299F, 0.587F, 0.114F, 0.0F, 0.0F, 0.299F, 0.587F, 0.114F, 0.0F, 0.0F, 0.299F, 0.587F, 0.114F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 1.0F, 0.0F};
+        ColorMatrixColorFilter colorMatrixFilter = new ColorMatrixColorFilter(matrix);
+        paint.setColorFilter((ColorFilter) colorMatrixFilter);
+        canvas.drawBitmap(image.getBitmap(), 0.0F, 0.0F, paint);
+
+        // Getting the pixels from the generated grayscale image
+        int w = bmpGrayscale.getWidth();
+        int h = bmpGrayscale.getHeight();
+        int[] intValues = new int[w * h];
+        bmpGrayscale.getPixels(intValues, 0, w, 0, 0, w, h);
+        // Shape with one channel
+        int[] shape = new int[]{1, h, w, 1};
+
+        switch (image.getDataType()) {
+            case UINT8:
+
+                // Create byte array and use one of the 3 identical channels
+                byte[] byteArr = new byte[w * h * 1];
+                for (int i = 0, j = 0; i < intValues.length; i++) {
+                    byteArr[j++] = (byte) (intValues[i] & 0xff);
```

Comment:
Is it necessary to use a different J here? It looks like j is always equal to i.

Reply:
Tested but it is not working without J. So I kept it that way. I used the same implementation as inside ImageConversions.java file